vspace/src/main/java/edu/asu/diging/vspace/web/staff/api/ImagesSearchFullApiController.java

```diff
@@ -0,0 +1,94 @@
+package edu.asu.diging.vspace.web.staff.api;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.StringUtils;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.model.ImageCategory;
+import edu.asu.diging.vspace.core.model.SortByField;
+import edu.asu.diging.vspace.core.model.impl.VSImage;
+import edu.asu.diging.vspace.core.services.IImageService;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Controller
+public class ImagesSearchFullApiController {
+
+    @Autowired
+    private IImageService imageService;
+
+    @Autowired
+    private ISpaceManager spaceManager;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping("/staff/images/search/full/{page}")
+    public String imageSearchDescription(@PathVariable String page,
+            @RequestParam(value = "searchText", required = false) String searchTerm, 
+            @RequestParam(value = "imageCat", required = false) String imageCategory,
+            @RequestParam(value = "sort", required = false) String sortedBy,
+            @RequestParam(value = "order", required = false) String order,
+            Model model, RedirectAttributes attributes) {
+        int pageNo;
+        page = StringUtils.isEmpty(page) ? "1" : page;
+        try {
+            pageNo = Integer.parseInt(page);
+        } catch (NumberFormatException numberFormatException) {
+            pageNo = 1;
+        }
+
+        ImageCategory category = null;
+        if (imageCategory!=null && imageCategory!="") {
+            try {
+                category = ImageCategory.valueOf(imageCategory);
+            } catch (IllegalArgumentException e) {
+                logger.error("Wrong argument for image category", e);
+            }
+        }
+        model.addAttribute("totalPages", category == null ? imageService.getTotalPagesOnSearchText(searchTerm)
```

Comment:
see my comment on `imageService`. the page object has the total pages information, so there shouldn't be another db call needed